src/main/java/hudson/tasks/junit/util/LinearRegression.java

```diff
@@ -0,0 +1,177 @@
+    }
+
+   /**
+     * Returns the <em>y</em>-intercept &alpha; of the best of the best-fit line <em>y</em> = &alpha; + &beta; <em>x</em>.
+     *
+     * @return the <em>y</em>-intercept &alpha; of the best-fit line <em>y = &alpha; + &beta; x</em>
+     */
+    public double intercept() {
+        return intercept;
+    }
+
+   /**
+     * Returns the slope &beta; of the best of the best-fit line <em>y</em> = &alpha; + &beta; <em>x</em>.
+     *
+     * @return the slope &beta; of the best-fit line <em>y</em> = &alpha; + &beta; <em>x</em>
+     */
+    public double slope() {
+        return slope;
+    }
+
+   /**
+     * Returns the coefficient of determination <em>R</em><sup>2</sup>.
+     *
+     * @return the coefficient of determination <em>R</em><sup>2</sup>,
+     *         which is a real number between 0 and 1
+     */
+    public double R2() {
+        return r2;
+    }
+
+   /**
+     * Returns the standard error of the estimate for the intercept.
+     *
+     * @return the standard error of the estimate for the intercept
+     */
+    public double interceptStdErr() {
+        return Math.sqrt(svar0);
+    }
+
+   /**
+     * Returns the standard error of the estimate for the slope.
+     *
+     * @return the standard error of the estimate for the slope
+     */
+    public double slopeStdErr() {
+        return Math.sqrt(svar1);
+    }
+
+   /**
+     * Returns the expected response {@code y} given the value of the predictor
+     * variable {@code x}.
+     *
+     * @param  x the value of the predictor variable
+     * @return the expected response {@code y} given the value of the predictor
+     *         variable {@code x}
+     */
+    public double predict(double x) {
+        return slope*x + intercept;
+    }
+
+   /**
+     * Returns a string representation of the simple linear regression model.
+     *
+     * @return a string representation of the simple linear regression model,
+     *         including the best-fit line and the coefficient of determination
+     *         <em>R</em><sup>2</sup>
+     */
+    public String toString() {
+        StringBuilder s = new StringBuilder();
+        s.append(String.format("%.2f n + %.2f", slope(), intercept()));
+        s.append("  (R^2 = " + String.format("%.3f", R2()) + ")");
+        return s.toString();
+    }
+
+}
+
+/******************************************************************************
+ *  Copyright 2002-2022, Robert Sedgewick and Kevin Wayne.
+ *
+ *  This file is part of algs4.jar, which accompanies the textbook
+ *
+ *      Algorithms, 4th edition by Robert Sedgewick and Kevin Wayne,
+ *      Addison-Wesley Professional, 2011, ISBN 0-321-57351-X.
+ *      http://algs4.cs.princeton.edu
+ *
+ *
+ *  algs4.jar is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
```

Comment:
I'm a bit worried about this, @jenkinsci/board any experience with GPL v3 and if its suitable for here?

I've read:
https://www.jenkins.io/project/governance/#3rd-party-library-licenses-in-the-core

but GPL v3 isn't covered there

Reply:
If it's a major problem and there are no suggestions for alternative libs, then we could disable it and work on re-enabling it via another PR. I tried to find something that is free, but ultimately it was an afterthought for which I had little time left.

Reply:
if this PR was broken down into multiple PRs then it would make something like this easier rather than holding the whole thing up